### DISCUSSION TITLE
Wallet Connect for Embedded Systems

### DISCUSSION THREAD

**Discussion by JHanuja** [WalletConnect/walletconnect-monorepo#3438] [Ideas]
Hello,

I wanted to ask if it makes sense to create a Wallet Connect library which can be run on a microcontroller. So in the end the microcontroller would show the qr code on it's display and after connecting with the user's wallet there could be actions completed via the interface of the microcontroller which then trigger smart contract transactions. 

I would be thankful for a reply.

**Comment by glitch-txs** [comment]
sign API or Pairing API should be useful here, not sure which language uses though
https://docs.walletconnect.com/2.0/api/sign/overview
